prepare/processors/processors.py

```diff
@@ -369,3 +371,35 @@
     "processors.extract_arena_hard_numerical_judgment",
     overwrite=True,
 )
+
+add_to_catalog(
+    SequentialOperator(
+        steps=[
+            InferDictsToBinaryLogprobs(
+                binary_class_names=(
+                    "No",
+                    "Yes",
+                ),
+                field="prediction",
+                process_every_value=False,
+            ),
+        ]
+    ),
+    "processors.infer_logprobs_to_yes_no_probs",
+)
+
+add_to_catalog(
+    SequentialOperator(
+        steps=[
+            InferDictsToBinaryLogProbsLastToken(
+                binary_class_names=(
+                    "No",
+                    "Yes",
+                ),
+                field="prediction",
+                process_every_value=False,
```

Comment:
Notice that we have a field called "options" assigned by many templates (such as the classification templates and multiple choice templates) with the options for completion based on the verbalization of the template. This is a safer way to do it because sometimes when you switch the template it's now "True" "False" and not "Yes" "No" and it's hard to pass it on. That's why so far all the verbalization reposibilty was in the template and externalizing verbalization to anywhere else in the pipeline can lead to issues. If you need help to integrate with the current mechanism of the "options" field you can either contact me or @perlitz who did an extensive use of it in the past specifically in the context of logprobs.

Reply:
@elronbandel Thanks, but I am not sure I follow how this post-processor is related to the template fields. We want this to be used on the inference engine predictions, across many templates for which this pattern would be applicable.
BTW, there are some other processors like this that have specific strings they expect, for example YesToOneElseZero, StanceToProCon

Reply:
I agree , it need not use a task "options" field.
A processor is tied to a template and not necessary to a task.

Reply:
The "options" field (which is not a task field but an internal unitxt field)  was our way to use it internally for getting log probs. It takes into account template nuances such as whether the model is expected to output the choice numeral (a. 1. etc) in multiple choice, and if we have target prefix. All I suggest is that you fully understand the current mechanism as it was developed very carefully.

Reply:
Yes. I was not too familiar with the option mechanism (we should document it).

Looking at the code. It's not used in  unitxt log prob inference but in the internal package inference.
"options" are  added to task/data  in the MultipleChoiceTemplate (if they are not already in the task).

In any case, it's a different scenario in the LLM as judge, where the user defines a template

"Is the following sentence correct? Answer 'with one word (Yes' or 'No')

'The sky is blue'
"

The model returns

"Yes it is."

And the post processor wants to return the probability of "Yes" vs "No" in the relevant token, (the first) so it converts the text into a float score.

Perhaps if force the model to use a MultipleChoiceTemplate, it would have the options fields - but do we want this for LLM as Judge? It seems cumbersome.